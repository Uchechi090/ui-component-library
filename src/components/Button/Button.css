.btn {
  border: none;
  border-radius: 15px;
  background-color: #a0006d;
  color: white;
  padding: 16px 32px;
  text-align: center;
  text-decoration: none;
  font-size: 16px;
  margin: 4px 2px;
  transition-duration: 0.4s;
  cursor: pointer;
}

.btn:hover {
  filter: brightness(85%);
  transition: 0.3s;
}

.btn:active {
  outline: 3px solid #e89717;
}

.disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.large {
  padding: 16px 55px;
}

.small {
  padding: 8px 10px;
  border-radius: 11px;
  height: 37px;
  align-self: center;
}

.primary {
  background-color: #0c6efd;
}

.secondary {
  background-color: #6c757d;
  padding: 16px 25px;
}
