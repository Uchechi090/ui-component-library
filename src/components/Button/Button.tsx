import React from "react";

import "./Button.css";

interface ButtonProps {
  label: string;
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  style?: object;
  className?: string | null;
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({
  label,
  onClick,
  style,
  className,
  disabled,
}) => {
  return (
    <button
      data-testid="btn"
      type="button"
      className={`btn ${className} ${disabled && 'disabled'}`}
      style={style}
      onClick={(e) => onClick(e)}
      disabled={disabled}
    >
      {label}
    </button>
  );
};

export default Button;
