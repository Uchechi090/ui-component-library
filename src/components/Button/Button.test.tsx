import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "./Button";
import Button from "./Button";

describe("<Button />", () => {
  const mockClickValue = jest.fn();
  const stubbedLabelValue = "Button";
  const stubbedStyleValue = {
    fontSize: "12px",
  };

  it("should show props", () => {
    const { getByTestId } = render(
      <Button
        label={stubbedLabelValue}
        onClick={mockClickValue}
        style={stubbedStyleValue}
      />
    );

    expect(getByTestId("btn").textContent).toBe("Button");
    expect(getByTestId("btn").style).toContain("font-size");

    fireEvent.click(getByTestId("btn"));
    expect(mockClickValue).toBeCalledTimes(1);
  });

  it("should contain classname for large", () => {
    const { getByTestId, asFragment } = render(
      <Button
        label={stubbedLabelValue}
        onClick={mockClickValue}
        className={"large"}
      />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("btn").className).toContain("large");
  });
  it("should contain classname for small", () => {
    const { getByTestId, asFragment } = render(
      <Button
        label={stubbedLabelValue}
        onClick={mockClickValue}
        className={"small"}
      />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("btn").className).toContain("small");
  });
  it("should contain classname for primary", () => {
    const { getByTestId, asFragment } = render(
      <Button
        label={stubbedLabelValue}
        onClick={mockClickValue}
        className={"primary"}
      />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("btn").className).toContain("primary");
  });
  it("should contain classname for secondary", () => {
    const { getByTestId, asFragment } = render(
      <Button
        label={stubbedLabelValue}
        onClick={mockClickValue}
        className={"secondary"}
      />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("btn").className).toContain("secondary");
  });
  it("should show disabled", () => {
    const { getByTestId, asFragment } = render(
      <Button
        label={stubbedLabelValue}
        onClick={mockClickValue}
        disabled={true}
      />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("btn").className).toContain("disabled");
  });
});
