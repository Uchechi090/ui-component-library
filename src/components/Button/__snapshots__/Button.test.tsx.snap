// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should contain classname for large 1`] = `
<DocumentFragment>
  <button
    class="btn large undefined"
    data-testid="btn"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`<Button /> should contain classname for primary 1`] = `
<DocumentFragment>
  <button
    class="btn primary undefined"
    data-testid="btn"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`<Button /> should contain classname for secondary 1`] = `
<DocumentFragment>
  <button
    class="btn secondary undefined"
    data-testid="btn"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`<Button /> should contain classname for small 1`] = `
<DocumentFragment>
  <button
    class="btn small undefined"
    data-testid="btn"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`<Button /> should show disabled 1`] = `
<DocumentFragment>
  <button
    class="btn undefined disabled"
    data-testid="btn"
    disabled=""
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
